wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift

```diff
@@ -255,6 +255,37 @@ final class SettingsTableViewController: SettingsBaseTableViewController {
         fatalError("Cannot dequeue cell for index path \(indexPath) and cellDescriptor \(cellDescriptor)")
     }
 
+    func tableView(_ tableView: UITableView, shouldShowMenuForRowAt indexPath: IndexPath) -> Bool {
+        let sectionDescriptor = sections[(indexPath as NSIndexPath).section]
+        let cellDescriptor = sectionDescriptor.visibleCellDescriptors[(indexPath as NSIndexPath).row]
+        return cellDescriptor.canCopy
+    }
+
+    func tableView(_ tableView: UITableView, contextMenuConfigurationForRowAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        let sectionDescriptor = sections[(indexPath as NSIndexPath).section]
+        let cellDescriptor = sectionDescriptor.visibleCellDescriptors[(indexPath as NSIndexPath).row]
+
+        guard cellDescriptor.canCopy else {
+            return nil
+        }
+
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { _ in
+          let copy = UIAction(title: "Copy", image: UIImage(systemName: "doc.on.doc")) { _ in
+            if let cell = tableView.cellForRow(at: indexPath) as? SettingsCellType {
```

Comment:
Do we need to access the cell (View) directly to get the data? Or is there some possibility here in the view controller to access the domain data model?

Reply:
I can downcast the cellDescriptor to SettingsExternalScreenCellDescriptor and retrieve via the previewGenerator...

or maybe change `canCopy` to `copiableString?` and retrieve from there, I'll try

Reply:
fixed